build-logic/packaging/src/main/kotlin/bisq/gradle/packaging/JLinkTask.kt

```diff
@@ -27,46 +31,103 @@ abstract class JLinkTask : DefaultTask() {
-
-        val jLinkPath = jdkDirectory.asFile.get().toPath().resolve("bin").resolve("jlink")
-        val processBuilder = ProcessBuilder(
+        if (outputDirectoryFile.exists() && outputDirectoryFile.listFiles()?.isNotEmpty() == true) {
+            // In case you want to run it manually, this will stop the plugin from running the task
+            logger.lifecycle("custom jre already created, skipping - see: ${jdkDirectory.get()}")
+        } else {
+            if (!outputDirectoryFile.canWrite()) {
+                throw IllegalStateException("Output directory is not writable: ${outputDirectoryFile.absolutePath}")
+            }
+            outputDirectoryFile.deleteRecursively()
+
+            logger.lifecycle("JDK path: ${jdkDirectory.get()}")
+
+            // Construct the jlink command
+            val jLinkPath = jdkDirectory.asFile.get().toPath().resolve("bin").resolve("jlink")
+            val processBuilder = ProcessBuilder(
                 jLinkPath.toAbsolutePath().toString(),
-
+                "--verbose",
                 "--add-modules", parseUsedJavaModulesFromJDepsOutput(),
-
+                "--include-locales=en,cs,de,es,it,pcm,pt-BR",
                 "--strip-native-commands",
                 "--no-header-files",
                 "--no-man-pages",
-                "--strip-debug",
-
+                "--compress=2",
+//            "--strip-debug", // makes jlink fail in some platforms like Ubuntu linux
+//            "--add-reads", "java.base=ALL-UNNAMED",
                 "--output", outputDirectoryFile.absolutePath
-        )
-
-        if (javaModulesDirectory.isPresent) {
-            val commands = processBuilder.command()
-            commands.add("--module-path")
-            commands.add(javaModulesDirectory.asFile.get().absolutePath)
-        }
-
-        processBuilder.inheritIO()
-
-        val process = processBuilder.start()
-        process.waitFor(2, TimeUnit.MINUTES)
-
-        val isSuccess = process.exitValue() == 0
-        if (!isSuccess) {
-            throw IllegalStateException("jlink couldn't create custom runtime.")
+            )
+            logger.lifecycle("Executing jlink command: ${processBuilder.command().joinToString(" ")}")
+
+            // Add module path if specified
+            if (javaModulesDirectory.isPresent) {
+                val commands = processBuilder.command()
+                commands.add("--module-path")
+                commands.add(javaModulesDirectory.asFile.get().absolutePath)
+                logger.lifecycle("Executing jlink command: ${commands.joinToString(" ")}")
+            }
+
+            logger.lifecycle("Preparing process builder..")
+            processBuilder.inheritIO()
+
+            logger.lifecycle("Starting process builder..")
+            val process = processBuilder.start()
+            val reader =
+                BufferedReader(InputStreamReader(process.inputStream))
+            while ((reader.readLine()) != null) {
```

Comment:
Maybe wrap it into a if(isWindows()) branch?

Reply:
it doesn't really affect the other platforms, but sure I'll have a look when testing removing javafx.web ;)